Roman Smirnov M3237

IEclipseI opened the pull request:
Смирнов Роман M3237

rvem commented:
![image](https://user-images.githubusercontent.com/14182218/55975053-c6648300-5c91-11e9-86ea-0a3119a01a32.png)
После ctrl+d шеллу становится плохо
В остальном все ок

rvem commented:
![image](https://user-images.githubusercontent.com/14182218/55975053-c6648300-5c91-11e9-86ea-0a3119a01a32.png)
После ctrl+d шеллу становится плохо
В остальном все ок

rvem commented:
![image](https://user-images.githubusercontent.com/14182218/55975053-c6648300-5c91-11e9-86ea-0a3119a01a32.png)
После ctrl+d шеллу становится плохо
В остальном все ок

IEclipseI commented:
> После ctrl+d шеллу становится плохо
> В остальном все ок

fixed, also fixed bug with nameless env variable(I failed with return in previous version)

ubique commented:
Зачтено.
